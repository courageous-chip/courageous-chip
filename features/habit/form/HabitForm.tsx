import { gql } from "@apollo/client";
import { useTheme } from "@react-navigation/native";
import React, { FC } from "react";
import { ScrollView, StyleSheet } from "react-native";

import { Button } from "../../ui/form/Button";
import { TextField } from "../../ui/form/TextField";
import { ListHabitsQuery } from "../list/__generated__/ListHabitsQuery";

type Props = { id?: ListHabitsQuery["habits"][0]["id"] };

export const HabitForm: FC<Props> = function ({ id }) {
  const {
    colors: { background: backgroundColor },
  } = useTheme();

  return (
    <ScrollView style={[styles.container, { backgroundColor }]}>
      <TextField
        marginBottom={styles.nameInput.marginBottom}
        placeholder="ðŸ¦· Floss Teeth"
      />
      <Button marginBottom={styles.saveButton.marginBottom} text="ðŸ’¾" />
      {id ? <Button text="ðŸ—‘" /> : null}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: { paddingHorizontal: 20, paddingVertical: 10 },
  nameInput: { marginBottom: 20 },
  saveButton: { marginBottom: 10 },
});

const CREATE_HABIT_MUTATION = gql`
  mutation CreateHabitMutation($habit: CreateHabitInput!) {
    createHabit(habit: $habit) {
      id
      name
    }
  }
`;

const DELETE_HABIT_MUTATION = gql`
  mutation DeleteHabitMutation($habit: DeleteHabitInput!) {
    deleteHabit(habit: $habit) {
      id
    }
  }
`;

const SHOW_HABIT_QUERY = gql`
  query ShowHabitQuery($id: ID!) {
    habit(id: $id) {
      id
      name
    }
  }
`;

const UPDATE_HABIT_MUTATION = gql`
  mutation UpdateHabitMutation($habit: UpdateHabitInput!) {
    updateHabit(habit: $habit) {
      id
      name
    }
  }
`;
