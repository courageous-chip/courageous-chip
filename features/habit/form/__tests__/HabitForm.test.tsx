import { render, wait } from "@testing-library/react-native";
import React from "react";

import { HabitForm } from "../HabitForm";

describe("<HabitForm />", () => {
  it("renders as empty", async () => {
    const { baseElement, getByText } = render(<HabitForm />);

    await wait(() => expect(getByText("🤷‍♀️")));

    expect(baseElement).toMatchSnapshot();
  });
});
