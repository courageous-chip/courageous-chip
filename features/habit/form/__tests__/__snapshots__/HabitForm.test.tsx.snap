// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitForm /> renders as empty 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 72,
          }
        }
      >
        🤷‍♀️
      </Text>
    </View>
  </View>
</View>
`;
